File: src/dataset/javascript/trinary_exercism-ThomasZumsteg/gemini/GeminiFlash_trinary_v4.js
var Trinary = function(digits) {
	this.digits = digits;
};

Trinary.prototype.toDecimal = function() {
	let decimal = 0;
	const digits = this.digits;
	const len = digits.length;

	for (let i = 0; i < len; i++) {
		const digit = digits[len - 1 - i];
		if (digit >= '0' && digit <= '2') {
			decimal += digit * (3 ** i);
		} else {
			return 0;
		}
	}

	return decimal;
};

module.exports = Trinary;

File: src/dataset/Java/trinary_exercism-ThomasZumsteg/gemini/GeminiFlash_Trinary_v4.java
public class Trinary {
    public static int toDecimal(String trinary) {
        if (trinary == null || trinary.isEmpty()) {
            return 0;
        }

        int decimal = 0;
        int length = trinary.length();

        for (int i = 0; i < length; i++) {
            char c = trinary.charAt(i);
            int digit = c - '0';

            if (digit >= 0 && digit <= 2) {
                decimal = decimal * 3 + digit;
            } else {
                return 0;
            }
        }

        return decimal;
    }
}